src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts

```diff
@@ -1162,3 +1178,136 @@ export class ObjectSettingWidget extends AbstractListSettingWidget<IObjectDataIt
 		};
 	}
 }
+
+interface IBoolObjectSetValueOptions {
+	settingKey: string;
+}
+export interface IBoolObjectDataItem {
+	key: string;
+	value: boolean;
+	description?: string;
+}
+
+export class BoolObjectSettingWidget extends AbstractListSettingWidget<IBoolObjectDataItem> {
+	private currentSettingKey: string = '';
+
+	override setValue(listData: IBoolObjectDataItem[], options?: IBoolObjectSetValueOptions): void {
+		if (isDefined(options) && options.settingKey !== this.currentSettingKey) {
+			this.model.setEditKey('none');
+			this.model.select(null);
+			this.model.setEditOnly(true);
+			this.currentSettingKey = options.settingKey;
+		}
+
+		super.setValue(listData);
+	}
+
+	isItemNew(item: IBoolObjectDataItem): boolean {
+		return !item.key && !item.value;
+	}
+
+	protected getEmptyItem(): IBoolObjectDataItem {
+		return {
+			key: '',
+			value: false
+		};
+	}
+
+	protected getContainerClasses() {
+		return ['setting-list-object-widget'];
+	}
+
+	protected getActionsForItem(item: IBoolObjectDataItem, idx: number): IAction[] {
+		return [];
+	}
+
+	protected override isAddButtonVisible(): boolean {
+		return false;
+	}
+
+	protected override renderHeader() {
+		return undefined;
+	}
+
+	protected renderItem(item: IBoolObjectDataItem): HTMLElement {
+		// return empty object, since we always render in edit mode
+		const rowElement = $('.setting-list-row');
+		return rowElement;
+	}
+
+	protected renderEdit(item: IBoolObjectDataItem, idx: number): HTMLElement {
+		const rowElement = $('.setting-list-edit-row.setting-list-object-row.setting-item-bool');
+
+		const changedItem = { ...item };
+		const onValueChange = (newValue: boolean) => {
+			changedItem.value = newValue;
+			this.handleItemChange(item, changedItem, idx);
+		};
+		const { element, widget: checkbox } = this.renderEditWidget(changedItem.value, onValueChange);
+		rowElement.appendChild(element);
+
+		const valueElement = DOM.append(rowElement, $('.setting-list-object-value'));
+		const valueTitleElement = DOM.append(valueElement, $('.value-title'));
+		const { label } = settingKeyToDisplayFormat(changedItem.key);
```

Comment:
I am actually not crazy about putting the key in the word format. In similar places we show the verbatim key (like enums) and I prefer this. The wordifying doesn't work well in some cases and I'd actually like to replace it with more customizable setting labels.

Reply:
Reverted it back to using the keys as-is.
Customizable setting labels can be a feature request for next iteration.